pkg/storage/mvcc_value.go

```diff
@@ -216,6 +237,50 @@ func EncodeMVCCValue(v MVCCValue) ([]byte, error) {
 	return buf, nil
 }
 
+// EncodeMVCCValueToBuf encodes an MVCCValue into its Pebble
+// representation. See the comment on MVCCValue for a description of
+// the encoding scheme.
+//
+// If extended encoding is required, the given buffer will be used. If
+// the provided buffer is not large enough an error is
+// returned. Callers can use (MVCCValue).ExtendedEncodingSize() to
+// determine the required buffer size before calling.
+//
+// TODO(ssd): Dedup with EncodeMVCCValue.
+func EncodeMVCCValueToBuf(v MVCCValue, buf []byte) ([]byte, error) {
```

Comment:
I've opted to duplicate this code for now to avoid touching EncodeMVCCValue at all.